include/deal.II/grid/manifold_lib.h

```diff
@@ -345,6 +345,8 @@ template <int dim, int spacedim = dim>
 class CylindricalManifold : public ChartManifold<dim,spacedim,3>
 {
 public:
+  static_assert(spacedim==3, "This class is only implemented for spacedim==3!");
```

Comment:
i actually removed it at some point because it makes life hard with dimension independent programming. Say I have a code which can be run in 2D and 3D and according to parameter file I optionally use `CylindricalManifold` in 3D only.

Reply:
Hmm... I don't really like to just get a undefined reference if one tries to create a `CylindricalManifold<2>` object. I would just provide an implementation by deriving from `StraightManifold` directly. I'll remove this `static_assert` and address the issue in a different PR.

Reply:
Or you remove the `static_assert` and just add a regular `Assert` -- this way, the code compiles in a dimension-independent way, but you can't use the class in 2d.